de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/controller/RepoManagementController.java

```diff
@@ -46,4 +46,15 @@ public void updateRepository(RepositoryConfiguration repoConfiguration) {
 	public Set<String> getAllProjectNames() {
 		return RepoRegistry.getInstance().getRepositories().keySet();
 	}
+	
+	/**
+	 * If a project with the name from configuration exists, it is updated, otherwise it is created.
+	 */
+	public void addOrUpdateRepository(RepositoryConfiguration repoConfiguration) throws URISyntaxException {
+		if (RepoRegistry.getInstance().getRepositories().containsKey(repoConfiguration.getProjectName())) {

```

Comment:
Any reason why we cant use the `getAllProjectNames().contains(...) `anymore instead of `RepoRegistry.getInstance().getRepositories().containsKey(...)` ?

Reply:
sure, we can